src/main/java/org/dependencytrack/parser/cyclonedx/CycloneDXVexImporter.java

```diff
@@ -81,6 +80,13 @@ public void applyVex(final QueryManager qm, final Bom bom, final Project project
         }
     }
 
+    private boolean shouldAuditVulnerability(org.cyclonedx.model.vulnerability.Vulnerability bomVulnerability, Vulnerability dtVulnerability) {
+        boolean result = true;
+        result = result && dtVulnerability.getVulnId().equals(bomVulnerability.getId());
+        result = result && (bomVulnerability.getSource() == null || dtVulnerability.getSource().equalsIgnoreCase(bomVulnerability.getSource().getName()));
```

Comment:
@syalioune I think the `bomVulnerability.getSource() == null` case should not cause the analysis to be applied (https://github.com/DependencyTrack/dependency-track/pull/2549#issuecomment-1493126424) and should be removed here.